packages/client/src/components/ReviewsDisplay/ReviewsDisplay.jsx

```diff
@@ -1,51 +1,68 @@
 import React, { useEffect, useState } from 'react';
-import {
-  Box,
-  List,
-  ListItem,
-  Divider,
-  ListItemAvatar,
-  Avatar,
-  Typography,
-  Button,
-  CircularProgress,
-} from '@mui/material';
+import { Box, List, Divider, Button, CircularProgress } from '@mui/material';
 
-import AddCircleIcon from '@mui/icons-material/AddCircle';
-import ThumbDownOffAltIcon from '@mui/icons-material/ThumbDownOffAlt';
-import ThumbUpIcon from '@mui/icons-material/ThumbUp';
-import { RatingStars } from '../RatingStars/RatingStars';
 import ArrowDropDownIcon from '@mui/icons-material/ArrowDropDown';
 import ArrowDropUpIcon from '@mui/icons-material/ArrowDropUp';
 import { MovieDetailsLayout } from '../../containers/MovieDetailsLayout/MovieDetailsLayout';
 import styled from '@emotion/styled';
 import { apiURL } from '../../apiURL';
+import { ReviewItem } from './ReviewItem';
+import { HeaderReview } from './HeaderReview';
+import { useUserContext } from '../../context/UserContext';
 
 export function ReviewsDisplay({ movieID }) {
+  const { user } = useUserContext();
   const [reviews, setReviews] = useState([]);
   const [isLoading, setIsLoading] = useState(false);
   const [showAllReviews, setShowAllReviews] = useState(false);
+  const [userReview, setUserReview] = useState(null);
 
   useEffect(() => {
     if (!movieID) return;
     const fetchReviewsList = async () => {
       setIsLoading(true);
-      if (movieID) {
-        try {
-          const response = await fetch(`${apiURL()}/reviews/movie/${movieID}`);
-          const data = await response.json();
-          setReviews(data);
-        } catch (error) {
-          setIsLoading(false);
-        }
+      try {
+        const response = await fetch(`${apiURL()}/reviews/movie/${movieID}`);
+        const data = await response.json();
+        setReviews(data);
+      } catch (error) {
+        setIsLoading(false);
       }
       setIsLoading(false);
     };
     fetchReviewsList();
   }, [movieID]);
+
+  useEffect(() => {
+    if (!movieID || !user) return;
+    const fetchUserReview = async () => {
+      try {
+        const userReviewResponse = await fetch(
+          `${apiURL()}/reviews/${movieID}/uid/${user.uid}`,
+        );
+        const userReviewData = await userReviewResponse.json();
+        if (Array.isArray(userReviewData)) {
```

Comment:
Hm, what are potential outcomes on a successful response here?

Reply:
so, for this one if there is no user.uid it will be undefined and it will cause error, so i have to make sure there is an array 
i edit the code now by checking if userReviewResponse.ok